core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/project/ProjectServiceClientImpl.java

```diff
@@ -423,4 +470,140 @@ private String normalizePath(String path) {
+    }
+
+    @Override
+    public Promise<ItemReference> createFolder(DevMachine devMachine, Path path) {
+        final String requestUrl =
+                devMachine.getWsAgentBaseUrl() + "/project/folder" + normalizePath(path.toString());
+
+        return asyncRequestFactory.createPostRequest(requestUrl, null)
+                                  .loader(loaderFactory.newLoader("Creating folder..."))
+                                  .send(dtoUnmarshaller.newUnmarshaller(ItemReference.class));
+    }
+
+    @Override
+    public Promise<Void> delete(DevMachine devMachine, Path path) {
+        final String requestUrl = devMachine.getWsAgentBaseUrl() + "/project" + normalizePath(path.toString());
+
+        return asyncRequestFactory.createRequest(DELETE, requestUrl, null, false)
+                                  .loader(loaderFactory.newLoader("Deleting project..."))
+                                  .send(noOpUnmarshaller);
+    }
+
+    @Override
+    public Promise<Void> copy(DevMachine devMachine, Path source, Path target, String newName, boolean overwrite) {
+        final String requestUrl =
+                devMachine.getWsAgentBaseUrl() + "/project/copy" + normalizePath(source.toString()) + "?to=" +
+                target.toString();
+
+        final CopyOptions copyOptions = dtoFactory.createDto(CopyOptions.class);
+        copyOptions.setName(newName);
+        copyOptions.setOverWrite(overwrite);
+
+        return asyncRequestFactory.createPostRequest(requestUrl, copyOptions)
+                                  .loader(loaderFactory.newLoader("Copying..."))
+                                  .send(noOpUnmarshaller);
+    }
+
+    @Override
+    public Promise<Void> move(DevMachine devMachine, Path source, Path target, String newName, boolean overwrite) {
+        final String requestUrl =
+                devMachine.getWsAgentBaseUrl() + "/project/move" + normalizePath(source.toString()) + "?to=" +
+                target.toString();
+
+        final MoveOptions moveOptions = dtoFactory.createDto(MoveOptions.class);
+        moveOptions.setName(newName);
+        moveOptions.setOverWrite(overwrite);
+
+        return asyncRequestFactory.createPostRequest(requestUrl, moveOptions)
+                                  .loader(loaderFactory.newLoader("Moving..."))
+                                  .send(noOpUnmarshaller);
+    }
+
+    @Override
+    public Promise<TreeElement> getTree(DevMachine devMachine, Path path, int depth, boolean includeFiles) {
+        final String requestUrl =
+                devMachine.getWsAgentBaseUrl() + "/project/tree" + normalizePath(path.toString()) + "?depth=" +
+                depth + "&includeFiles=" +
+                includeFiles;
+
+        return asyncRequestFactory.createGetRequest(requestUrl)
+                                  .header(ACCEPT, MimeType.APPLICATION_JSON)
+                                  .loader(loaderFactory.newLoader("Reading structure..."))
```

Comment:
structure of ? (I don't know if end user knows what is being done by seeing this message)